plugin-server/src/main/job-queues/job-queue-manager.ts

```diff
@@ -50,7 +58,35 @@ export class JobQueueManager implements JobQueue {
         }
     }
 
-    async enqueue(jobName: string, job: EnqueuedJob): Promise<void> {
+    async enqueue(jobName: string, job: EnqueuedJob, instrumentationContext?: InstrumentationContext): Promise<void> {
+        const jobType = 'type' in job ? job.type : 'buffer'
+        const jobPayload = 'payload' in job ? job.payload : job.eventPayload
+        const pluginServerMode = this.pluginsServer.PLUGIN_SERVER_MODE ?? 'full'
+        await instrument(
+            this.pluginsServer.statsd,
+            {
+                metricName: jobName === JobName.PLUGIN_JOB ? 'vm.enqueuePluginJob' : 'vm.enqueueBufferJob',
+                key: instrumentationContext?.key ?? '?',
+                tag: instrumentationContext?.tag ?? '?',
+                tags: { pluginServerMode, type: jobType },
```

Comment:
There's no reason to add pluginServerMode tag - https://github.com/PostHog/posthog/blob/master/plugin-server/src/utils/db/hub.ts#L82-L84